lib/pre/reqQueueDepth.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Restify. All rights reserved.
+
+'use strict';
+
+var assert = require('assert-plus');
+
+
+///--- API
+
+
+/**
+ * capture request queue depth.
+ * @public
+ * @function createReqQueueDepth
+ * @param {Object} opts an options object
+ * @param {Object} opts.server a restify server
+ * @returns {Function}
+ */
+function createReqQueueDepth(opts) {
+
+    assert.object(opts, 'opts');
+    assert.object(opts.server, 'opts.server');
+
+    if (typeof opts.server.getReqQueueDepth === 'function') {
+        throw new Error('server object already has `getReqQueueDepth()` ' +
+                        'method!') ;
+    }
+
+    var QUEUE_LENGTH = 0;
+
+    // create method on server
+    opts.server.getReqQueueDepth = function getReqQueueDepth() {
+        return QUEUE_LENGTH;
+    };
+
+    return function reqQueueDepth(req, res, next) {
+
+        // increase request queue depth
+        QUEUE_LENGTH += 1;
+
+        function _pop() {
+            QUEUE_LENGTH -= 1;
```

Comment:
You should probably add an assert here and log a warn, if QUEUE_LENGTH is ever potentially < 0, that means there's a bug somewhere, probably best to keep length at 0 and not go negative.